### DISCUSSION TITLE
Discontinuous boundary conditions

### DISCUSSION THREAD

**Discussion by ke7kto** [idaholab/moose#17363] [Q&A General] [answered]
I'm solving a system of coupled equations (heat + diffusion) for a system of concentric rings. There is a gap between the rings, which I would rather not model, that leads to a difference in temperature between the inner-right and outer-left boundaries of the rings. I can specify these as a Dirichlet boundary condition (time-dependent function). I also have a relationship that defines the concentration at either side of the interface as a function of the temperature. 

Do I need to use XFEM to model something like this?  If so, how would I specify a function-based Dirichlet boundary condition, and how would I specify an equilibrium relationship across the boundary?

**Comment by aeslaughter** [comment]
This sounds like something that our [Constraints System](https://mooseframework.inl.gov/syntax/Constraints/index.html) might handle. I have not used it, perhaps @lindsayad or @bwspenc could verify that I am understanding correctly.

**Reply by ke7kto** [reply]
That looks right. I'll update if/when I get it working.

**Comment by dschwen** [comment, marked as answer]
I think you're looking for GapHeatTransfer: https://mooseframework.inl.gov/source/bcs/GapHeatTransfer.html

**Reply by ke7kto** [reply]
That might be a good starting point. Still new here so my terminology might be off, but I would almost describe temperature as an auxiliary variable. I still need to calculate the temperature profile, but I have time-dependent values of the temperature at each edge of each gap. I need to calculate the equilibrium concentration across the gap, assuming fast gas-phase transport. I do have an equation for that. Would you recommend me starting with GapHeatTransfer and extending it?
